Description: Change ultimate fallback editor for fc
 Hardly anybody knows how to use 'ed' and it is not installed
 by default. Changing to 'ex' which at least is similar
 and acts as the gateway to the default editor.
Author: Nicholas Bamber <nicholas@periapt.co.uk>
Bug-Debian: https://bugs.debian.org/633524
Forwarded: not-needed
Last-Update: 2015-11-02
--- a/src/cmd/ksh93/features/cmds
+++ b/src/cmd/ksh93/features/cmds
@@ -1,4 +1,4 @@
 cmd	newgrp,test,id,wc,cut,logname,universe,pfexec,tput
 pth	ed fail{
-	echo '#define _pth_ed	"ed" /* ed not found on standard PATH */'
+	echo '#define _pth_ed	"/usr/bin/ex" /* ed not found on standard PATH */'
 }end
--- a/src/cmd/ksh93/bltins/hist.c
+++ b/src/cmd/ksh93/bltins/hist.c
@@ -218,7 +218,7 @@
 	{
 		arg = (char*)e_defedit;
 		if(*arg!='/')
-			errormsg(SH_DICT,ERROR_exit(1),"ed not found set FCEDIT");
+			errormsg(SH_DICT,ERROR_exit(1),"ex not found set FCEDIT");
 	}
 #ifdef apollo
 	/*
--- a/src/cmd/ksh93/sh.1
+++ b/src/cmd/ksh93/sh.1
@@ -4355,7 +4355,7 @@
 .SM
 .B FCEDIT
 is not defined, then
-.B /bin/ed
+.B /usr/bin/ex
 is used.
 The edited command(s) is printed and re-executed upon
 leaving the editor unless you quit without writing.
@@ -6141,7 +6141,7 @@
 .SM
 .B FCEDIT
 (default
-.BR /bin/ed\^ )
+.BR /usr/bin/ex\^ )
 is used as the editor.
 When editing is complete, the edited command(s)
 is executed if the changes have been saved.
